dist/common/scripts/scylla_coredump_setup

```diff
@@ -62,8 +62,7 @@ ExternalSizeMax=1024G
 [Unit]
 Description=Save coredump to scylla data directory
 Conflicts=umount.target
-Before=scylla-server.service
-After=local-fs.target
+Before=local-fs.target scylla-server.service
```

Comment:
Why do we want to start coredump before the local filesystem? I don't understand.

Reply:
Then why do we have Before=scylla-server.service? It needs to be After too, no?

Reply:
I'm so confused now.

Reply:
```
If unit foo.service contains the setting Before=bar.service and both units are being started, bar.service's start-up is delayed
```

translated

```
If unit systemd-coredump.service contains the setting Before=local-fs.target and both units are being started, local-fs.target's start-up is delayed
```

Do we want to delay local-fs.target? I think not.

Reply:
> Do we want to delay local-fs.target? I think not.

@avikivity i disagree.

from `var-lib-systemd-coredump` 's perspective, it's safe to start even without `local-fs`. but the thing is, from `local-fs` 's perspective, we need to ensure that the service units depending on (i.e. `Wants` and `After`) `local-fs` can have access to local filesystems after `local-fs` is started. 

if we don't delay `local-fs.target` 's start-up, we cannot ensure that the service unit depending on `local-fs` can access `/var/lib/systemd/coredump` even if they are marked with `After=local-fs.target` and `Wants=local-fs.target`. because `local-fs.target` can start in parallel with `var-lib-systemd-coredump`, if we don't add `Before=local-fs.target` in `var-lib-systemd-coredump` .

Reply:
> Then why do we have Before=scylla-server.service? It needs to be After too, no?

no. it should be `Before=scylla-server.service`. and probably we should also add `WantedBy=scylla-server.service`.

Reply:
created #15984 to address this.